src/relentless/model/potential/bond.py

```diff
@@ -0,0 +1,425 @@
+import numpy
+
+from relentless import math
+
+from . import potential
+
+
+class BondParameters(potential.Parameters):
+    """Parameters for a bond potential."""
+
+    pass
+
+
+class BondPotential(potential.Potential):
+    r"""Abstract base class for a bond potential."""
+
+    pass
+
+
+class Harmonic(BondPotential):
+    r"""Harmonic bond potential.
+
+    .. math::
+
+        u(r) = \frac{k}{2} (r - r_0)^2
+
+    where :math:`r` is the distance between two bonded particles. The parameters
+    for each type are:
+
+    +-------------+--------------------------------------------------+-----------+
+    | Parameter   | Description                                      | Initial   |
+    +=============+==================================================+===========+
+    | ``k``       | potential constant                               |           |
+    +-------------+--------------------------------------------------+-----------+
+    | ``r0``      | rest length.                                     |           |
+    +-------------+--------------------------------------------------+-----------+
+
+    Parameters
+    ----------
+    type : str
+        Type.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`BondParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    Harmonic Bond::
+
+        >>> u = relentless.potential.bond.Harmonic("A")
+        >>> u.coeff["A"].update({'k': 1.0, 'r0': 1.0})
```

Comment:
I like to give examples with sensible choices. A spring constant of 1 is too small for most bonds.
```suggestion
        >>> u.coeff["A"].update({'k': 1000, 'r0': 1})
```